packages/api/src/graphql/mutation.public.ts

```diff
@@ -536,6 +538,49 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
           failureURL
         })
       }
+    },
+
+    checkInvoiceStatus: {
+      type: GraphQLPublicInvoice,
+      args: {
+        id: {type: GraphQLNonNull(GraphQLID)}
+      },
+      description:
+        'This mutation will check the invoice status and update with information from the paymentProvider',
+      async resolve(root, {id}, context) {
+        const {authenticateUser, dbAdapter, paymentProviders} = context
+        const {user} = authenticateUser()
+
+        const invoices = await dbAdapter.invoice.getInvoicesByUserID(user.id)
```

Comment:
I've noticed you check presence of item in those arrays returned from database. Do we expect null/undefined here?

Same on line `562`, 565`, `570`.

Reply:
In this case we don't. But there are methods that accept a list of ids and then return the found models. There it could be that some of the items could be `null`. In any case we use the same type `OptionalInvoice` and that's why i have to check.